presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -805,7 +805,9 @@ public ListeningExecutorService createResourceManagerExecutor(ResourceManagerCon
         // Worker session property providers
         MapBinder<String, WorkerSessionPropertyProvider> mapBinder =
                 newMapBinder(binder, String.class, WorkerSessionPropertyProvider.class);
-        mapBinder.addBinding("java-worker").to(JavaWorkerSessionPropertyProvider.class).in(Scopes.SINGLETON);
+        if (!featuresConfig.isExcludeJavaWorkerSessionProperties()) {
+            mapBinder.addBinding("java-worker").to(JavaWorkerSessionPropertyProvider.class).in(Scopes.SINGLETON);
+        }
         if (!serverConfig.isCoordinatorSidecarEnabled()) {
             mapBinder.addBinding("native-worker").to(NativeWorkerSessionPropertyProvider.class).in(Scopes.SINGLETON);
```

Comment:
Do we need the same for native-worker session properties?  Maybe the config could have a generic name to exclude properties that aren't relevant for that worker type.